framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -260,50 +260,58 @@ public Void scan(Tree tree, Void p) {
         return super.scan(tree, p);
     }
 
+    /**
+     * Type-check classTree. If a subclass overrides this method, if should first check if this
+     * class should be skipped by calling {@link BaseTypeChecker#shouldSkipDefs(ClassTree)}.
+     *
+     * @param classTree class to check
+     * @param p null
+     * @return null
+     */
     @Override
-    public Void visitClass(ClassTree node, Void p) {
-        if (checker.shouldSkipDefs(node)) {
-            // Not "return super.visitClass(node, p);" because that would
+    public Void visitClass(ClassTree classTree, Void p) {
+        if (checker.shouldSkipDefs(classTree)) {
+            // Not "return super.visitClass(classTree, p);" because that would
             // recursively call visitors on subtrees; we want to skip the
             // class entirely.
             return null;
         }
 
-        atypeFactory.preProcessClassTree(node);
+        atypeFactory.preProcessClassTree(classTree);
 
         AnnotatedDeclaredType preACT = visitorState.getClassType();
         ClassTree preCT = visitorState.getClassTree();
         AnnotatedDeclaredType preAMT = visitorState.getMethodReceiver();
         MethodTree preMT = visitorState.getMethodTree();
         Pair<Tree, AnnotatedTypeMirror> preAssCtxt = visitorState.getAssignmentContext();
-        visitorState.setClassType(atypeFactory.getAnnotatedType(node));
-        visitorState.setClassTree(node);
+        visitorState.setClassType(atypeFactory.getAnnotatedType(classTree));
+        visitorState.setClassTree(classTree);
         visitorState.setMethodReceiver(null);
         visitorState.setMethodTree(null);
         visitorState.setAssignmentContext(null);
 
         try {
-            if (!TreeUtils.hasExplicitConstructor(node)) {
-                checkDefaultConstructor(node);
+            if (!TreeUtils.hasExplicitConstructor(classTree)) {
```

Comment:
An alternative solution to this issue might be to extract this block (maybe a bit more or less) into a new method and then have that method as the common extension point that subclasses should override if they want to change behavior.
That way, most subclasses don't need to remember to also skip classes in certain situation, which this PR has to copy in 5 places.